connect/mirror/src/main/java/org/apache/kafka/connect/mirror/MirrorConnectorConfig.java

```diff
@@ -199,8 +199,8 @@
 
     protected static final String SOURCE_CLUSTER_PREFIX = MirrorMakerConfig.SOURCE_CLUSTER_PREFIX;
     protected static final String TARGET_CLUSTER_PREFIX = MirrorMakerConfig.TARGET_CLUSTER_PREFIX;
-    protected static final String PRODUCER_CLIENT_PREFIX = "producer.";
-    protected static final String CONSUMER_CLIENT_PREFIX = "consumer.";
+    protected static final String PRODUCER_CLIENT_PREFIX = SOURCE_CLUSTER_PREFIX + "producer.";
+    protected static final String CONSUMER_CLIENT_PREFIX = SOURCE_CLUSTER_PREFIX + "consumer.";
```

Comment:
These changes are going to break existing users. For example, I have connectors with a few settings prefixed with `consumer.`. I wonder if we could keep the old behaviour (even if partially broken) while adding the proper prefixes

Reply:
I'm not sure if I follow it. This prefix "source_cluster" is not user specified. It is prefixed somewhere else by mm2 (I can get the lines if you want later on). The config specified [here](https://github.com/apache/kafka/tree/trunk/connect/mirror#producer--consumer--admin-config-used-by-mm2) will actually be honored. It does not imply changes on the MirrorMaker config side at least. Are you talking about the case of running MirrorMaker in a connect cluster rather than as a dedicated cluster? I didn't check that one TBH. Maybe @ryannedolan has more insight on this one?

Reply:
Yes I'm running MM2 in a Connect cluster.

Reply:
Gotcha. I just peek at it again and you're right. It seems the proper fix would be to fix the way mm2 populates the MirrorConnectorConfig to avoid this change. WDYT @ryannedolan ?

Reply:
I have put a new fix that should handle both cases.

Reply:
It's unfortunate for admin we use `source.admin` as the prefix ...

So we'd be left with a configuration like:
```
"source.cluster.bootstrap.servers": "localhost:9092",
"source.cluster.security.protocol": "SASL_SSL",
"source.cluster.producer.some-producer-setting": 123
"source.cluster.consumer.some-consumer-setting": 123
"source.admin.some-admin-setting": 123
```

Reply:
Maybe not including `cluster` for all the client specific settings would be slightly better, WDYT?
```
"source.cluster.bootstrap.servers": "localhost:9092",
"source.cluster.security.protocol": "SASL_SSL",
"source.producer.some-producer-setting": 123,
"source.consumer.some-consumer-setting": 123,
"source.admin.some-admin-setting": 123
```

Reply:
For connect, you can still use (or the one you've shared).
```
"source.cluster.bootstrap.servers": "localhost:9092",
"source.cluster.security.protocol": "SASL_SSL",
"producer.some-producer-setting": 123
"consumer.some-consumer-setting": 123
"source.admin.some-admin-setting": 123
```

Yes, we could normalize that but I'm not touching admin config in this PR.

Reply:
> Maybe not including cluster for all the client specific settings would be slightly better, WDYT?

Somehow I missed this. I do not have a strong opinion on that but I do like the idea on normalizing it. But changing that will still break existing users though. How do you want to proceed?

Reply:
@mimaison just a friendly ping.

Reply:
I don't think the format mentioned in https://github.com/apache/kafka/pull/9313#discussion_r498298987 would break compatibility.

Reply:
For users running on a Connect cluster, that format is not supported right now:
```
source.producer.some-producer-setting: 123
```

The only supported is (unless I'm missing something):
```
"producer.some-producer-setting": 123
```

Reply:
I meant add support for that format. We obviously want to keep supporting the existing formats

Reply:
🤦 got it. I misunderstood it. Let me give it a try.

Reply:
I think the new code does what we want. Also tested a bin from this on my infra and it is working fine. Let me know if this works for you.